docs/proposals/templates-v2.md

```diff
@@ -0,0 +1,325 @@
+
+### Summary
+
+This can be done through a series of changes to the way that templates are implemented:
+
+1. Add a `${title}` and `${titleSlug}` template variable
+2. Add a `Foam: Create New Note` command and hotkey
+3. Change missing wikilinks to use the default template
+4. Add a metadata section to templates
+5. Add a replacement for `dateFormat`
+6. Add support for daily note templates
+7. Eliminate all `foam.openDailyNote` settings
+
+I've broken it out into these steps to show that the overall proposal can be implemented piecemeal in independent PRs that build on one another.
+
+### Add a `${title}` and `${titleSlug}` template variables
+
+When you use `Markdown Notes: New Note`, and give it a title, the title is formatted as a filename and used as the title in the resulting note.
+
+**Example:**
+
+Given the name `Living in a dream world` to `Markdown Notes: New Note`, the filename is `living-in-a-dream-world.md` and the file contents are:
+
+```markdown
+# Living in a dream world
+```
+
+When creating a note from a template in Foam, you should be able to use a `${title}` variable. If the template uses the `${title}` variable, the user will be prompted for a title when they create a note from a template.
+
+Example:
+
+Given this `.foam/templates/my_template.md` template that uses the `${title}` variable:
+
+```markdown
+# ${title}
+```
+
+When a user asks for a new note using this template (eg. `Foam: Create New Note From Template`), VSCode will first ask the user for a title then provide it to the template, producing:
+
+```markdown
+# Living in a dream world
+```
+
+There will also be a `${titleSlug}` variable made available, which will be the "slugified" version of the title (eg. `living-in-a-dream-world`). This will be useful in later steps where we want to template the filepath of a template.
+
+### Add a `Foam: Create New Note` command and hotkey
+
+Instead of using `Markdown Notes: New Note`, Foam itself will have a `Create New Note` command that creates notes using templates.
+
+This would open use the template found at `.foam/templates/new-note.md` to create the new note.
+
+`Foam: Create New Note` will offer the fastest workflow for creating a note when you don't need customization, while `Foam: Create New Note From Template` will remain to serve a fully customizable (but slower) workflow.
+
+#### Case 1: `.foam/templates/new-note.md` doesn't exist
+
+If `.foam/templates/new-note.md` doesn't exist, it behaves the same as `Markdown Notes: New Note`:
+  * it would ask for a note name and create the note in the current directory. It would open a note with the note name as the title.
+
+**Note:** this would use an implicit default template, making use of the `${title}` variable.
+
+#### Case 2: `.foam/templates/new-note.md` exists
+
+If `.foam/templates/new-note.md` exists:
+  * it asks for the note name and creates the note in the current directory
```

Comment:
what do you mean here by `name`: `title` or `filename`?

Reply:
Good point! I've gone through the document and removed all uses of `name`, changing them to either `title` or `filepath` as appropriate.